plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -3258,6 +3259,80 @@ public WriterScalingOptions getInsertWriterScalingOptions(ConnectorSession sessi
         return WriterScalingOptions.ENABLED;
     }
 
+    @Override
+    public Optional<ConnectorTableHandle> applyDelete(ConnectorSession session, ConnectorTableHandle handle)
+    {
+        DeltaLakeTableHandle tableHandle = (DeltaLakeTableHandle) handle;
+        if (tableHandle.getMetadataEntry().isChangeDataFeedEnabled().orElse(false)) {
+            // For tables with CDF enabled the DELETE operation can't be performed only on metadata files
+            return Optional.empty();
+        }
+        if (!tableHandle.getNonPartitionConstraint().isAll()) {
+            return Optional.empty();
+        }
+
+        return Optional.of(tableHandle);
+    }
+
+    @Override
+    public OptionalLong executeDelete(ConnectorSession session, ConnectorTableHandle handle)
+    {
+        DeltaLakeTableHandle tableHandle = (DeltaLakeTableHandle) handle;
```

Comment:
Please throw an exception when `isAppendOnly` returns true. `TestDeltaLakeDeleteCompatibility` should be updated to cover this secneario.